pkgdb/tests/lock-flake-installable.cc

```diff
@@ -0,0 +1,372 @@
+
+  return true;
+}
+
+bool
+test_names( const nix::ref<nix::EvalState> & state, const std::string & system )
+{
+  auto named
+    = flox::lockFlakeInstallable( state, system, localTestFlake + "#names" );
+
+  EXPECT_EQ( named.pname.value(), "hello" );
+  EXPECT_EQ( named.name, "explicit-name" );
+
+  return true;
+}
+
+bool
+test_version( const nix::ref<nix::EvalState> & state,
+              const std::string &              system )
+{
+  auto nonVersioned
+    = flox::lockFlakeInstallable( state, system, localTestFlake + "#hello" );
+
+  EXPECT( ! nonVersioned.version.has_value() );
+
+  auto versioned = flox::lockFlakeInstallable( state,
+                                               system,
+                                               localTestFlake + "#versioned" );
+
+  EXPECT_EQ( versioned.version.value(), "1.0" );
+
+  return true;
+}
+
+bool
+test_broken( const nix::ref<nix::EvalState> & state,
+             const std::string &              system )
+{
+  auto broken
+    = flox::lockFlakeInstallable( state, system, localTestFlake + "#broken" );
+
+  // with broken = true, the package does not even evaluate
+  EXPECT_EQ( broken.broken.value(), false );
+
+  return true;
+}
+
+bool
+test_unfree( const nix::ref<nix::EvalState> & state,
+             const std::string &              system )
+{
+  auto unfree
+    = flox::lockFlakeInstallable( state, system, localTestFlake + "#unfree" );
+
+  // with unfree = true, the package does not even evaluate
+  EXPECT_EQ( unfree.unfree.value(), false );
+
+  return true;
+}
+
+/* -------------------------------------------------------------------------- */
+
+int
+main( int argc, char * argv[] )
+{
+  int exitCode = EXIT_SUCCESS;
+#define RUN_TEST( ... ) _RUN_TEST( exitCode, __VA_ARGS__ )
+
+  nix::verbosity = nix::lvlWarn;
+  if ( ( 1 < argc ) && ( std::string_view( argv[1] ) == "-v" ) )  // NOLINT
+    {
+      nix::verbosity = nix::lvlDebug;
+    }
+
+  /* Initialize `nix' */
+  flox::NixState nstate;
+  auto           state = nstate.getState();
+
+  std::string system = nix::nativeSystem;
+
+  RUN_TEST( attrpathUsesDefaults, state, system );
+  RUN_TEST( flakerefOrigins, state, system );
+  RUN_TEST( locksUrl, state, system );
+  RUN_TEST( explicitOutputs, state, system );
+  RUN_TEST( resolvesToDefaultPackage, state, system );
+  RUN_TEST( systemAttributes, state );
+  RUN_TEST( licenseString, state, system );
+  RUN_TEST( licenseAttrs, state, system );
+  RUN_TEST( licenseListOfAttrs, state, system );
+  RUN_TEST( licenseNoLicense, state, system );
+  RUN_TEST( description, state, system );
+  RUN_TEST( names, state, system );
+  RUN_TEST( version, state, system );
+  RUN_TEST( broken, state, system );
+  RUN_TEST( unfree, state, system );
+
+  return exitCode;
+}
+
+
+/* -------------------------------------------------------------------------- *
+ *
+ *
+ *
+ * ========================================================================== */
```

Comment:
I like that we have a little bit of coverage for nixpkgs, but most test cases are with speedy to evaluate local cases ❤️